In this project we used the Bisection Method in order to find the zero of a function.<br/>

The Bisection Method works by taking an interval from [a, b]. For this problem lets say that it is between .01 and .99, as our interest rate will probably fall between that interval. The midpoint we will be c, the value between a and b. <br/>

If this midpoint is equal to zero than we have our midpoint and we exit the program.<br/>

If the product of f(a)  and f(c)the interval and get back a positive number we know that we are on the left side of our interval and we choose c to be our new a. After this, if we find our new c value is will be moved to the right. This reduces the right side of our interval and gets us closer the the zero<br/>

If the product our imputs for f(a) and f(c) are negative we reduce down the left side of our interval. We keep doing this until the differences between the inputs for f(a) and f(c) is less than our tolerence for error<br/>

We want to use this method to find the interest rate of a refrigerater. We want to find if it is between to save the money each month for the frige or to purchase the frige with credit:


Variables:<br/>
a = amount<br/>
p = principal<br/>
r = rate<br/>
m = months<br/>
n = number of times compounded<br/>

a = 4055.62<br/>
p = 1802.50<br/>
r = variable<br/>
m = 23<br/>
n = 12<br/>

Roo of Equation:

$$0 = n*((\frac{a}{p})^{\frac{1}{m}}- 1) - r$$

In [14]:
def bisection(a,b,f,err):
    c = (a+b)/2
    while c-a > err:
        if f(c) == 0:
            return n
        elif f(c)*f(a) > 0:
            a = c
        else:
            b = c
        c = (a+b)/2
    return c

In [15]:
def f(r):
    return 12*((4055.62/1802.50)**(1/23) - 1) - r

In [16]:
f(.01)

0.42064048655267494

In [17]:
f(.99)

-0.559359513447325

The root of f is between .01 and .99. That's what we will choose for our a and b

In [18]:
bisection(.01,.99,f,.0001)

0.4306750488281249

In [19]:
#Test that our formula is correct
1802.50*(1 + 0.4306750488281249/12)**(23)
#We get the correct total amount

4055.879362507174

In [20]:
1802.50/181.98
#It will take you ten months to save for the frige

9.904934608198703

In [21]:
4055.62 - 1802.50
#You will pay about 2000 dollars more if you work with the loan

2253.12

Frige Interest Rate: 43%<br/>

We find that instead of paying 23 monthly payments of 181.98, it is much better to save 181.98 every month until we get the 1802.50. It will only take 10 months and you do not take the risk of missing a payment.<br/>

You will also save more than 2000 dollars.<br/>

Link to product: https://www.rentacenter.com/appliances/refrigerators/maytag-stainless-20-cu-ft-french-door-refrigerator--mfw2055frz/p/100021764